lib/_http_common.js

```diff
@@ -290,11 +289,12 @@ exports._checkIsHttpToken = checkIsHttpToken;
  *  field-vchar    = VCHAR / obs-text
  **/
 function checkInvalidHeaderChar(val) {
-  val = '' + val;
+  if (typeof val !== 'string')
+    val = '' + val;
   for (var i = 0; i < val.length; i++) {
     const ch = val.charCodeAt(i);
-    if (ch === 9) continue;
-    if (ch <= 31 || ch > 255 || ch === 127) return true;
+    if (ch !== 9 && (ch <= 31 || ch > 255 || ch === 127))
```

Comment:
I don't understand, you're changing a `return` to a `continue`, that continue will go and hit the next value in `val` which will iterate it.

Mind explaining how it will continue iterating explicitly? Is that code around the code that was changed?